apps/web/src/components/settings/owner/OwnerList/index.tsx

```diff
@@ -17,26 +17,36 @@ import { TxModalContext } from '@/components/tx-flow'
 import ReplaceOwnerIcon from '@/public/images/settings/setup/replace-owner.svg'
 import DeleteIcon from '@/public/images/common/delete.svg'
 import type { AddressBook } from '@/store/addressBookSlice'
-
 import tableCss from '@/components/common/EnhancedTable/styles.module.css'
+import useResolvedOwners from '@/hooks/useResolvedOwners'
 
 export const OwnerList = () => {
   const addressBook = useAddressBook()
   const { safe } = useSafeInfo()
   const { setTxFlow } = useContext(TxModalContext)
+  const owners = useResolvedOwners()
 
   const rows = useMemo(() => {
-    const showRemoveOwnerButton = safe.owners.length > 1
+    const showRemoveOwnerButton = owners.length > 1
 
-    return safe.owners.map((owner) => {
+    return owners.map((owner) => {
       const address = owner.value
-      const name = addressBook[address]
+      let name = addressBook[address] || owner.name || undefined
```

Comment:
```suggestion
      const name = addressBook[address] || owner.name || undefined
```

This variable is never reassigned, so it can remain a constant.